scripts/migrate_project_mailing_lists.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+"""Create mailing lists for all top-level projects
+"""
+import logging
+import sys
+
+from modularodm import Q
+
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from website.models import Node
+from website.notifications.model import NotificationSubscription
+from website.notifications.utils import to_subscription_key
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+EVENT = 'mailing_list_events'
+SUBSCRIPTION_TYPE = 'email_transactional'
+
+def get_targets():
+    return Node.find()
+
+def migrate(dry_run=True):
+    successful_enables = []
+    successful_disables = []
+    unknown_failures = {}
+    nodes = get_targets()
+    ncount = len(nodes)
+    logger.info('Preparing to migrate {} nodes.'.format(ncount))
+    for node in nodes:
+        if node.is_registration or node.is_collection or node.is_deleted:
+            try:
+                logger.info('({0}/{1})Disabling mailing list for registration/dashboard {2}'.format(nodes.index(node)+1, ncount, node._id))
+                node.mailing_enabled = False    
+                node.save()
+                successful_disables.append(node._id)
+            except Exception as e:
+                unknown_failures[node._id] = e
+        else:
+            try:
+                subscription = NotificationSubscription(
+                    _id=to_subscription_key(node._id, EVENT),
+                    owner=node,
+                    event_name=EVENT
+                )
+                logger.info('({0}/{1})Enabling mailing list for node {2}'.format(nodes.index(node)+1, ncount, node._id))
```

Comment:
This line results in an error: `'MongoQuerySet' object has no attribute 'index'`. You can use `enumerate` to get the index for each iteration.